### PR TITLE
feat: Allow multiple responses for a unique request ID

### PR DESCRIPTION
**Pull request by alaney**
### Summary
This change allows multiple responses to be recorded for the same request.

For example, if an app makes a GET request to the endpoint `/todo/1` it will record the response within a response collection for the calculated request ID.

Any subsequent responses for the same GET requests will be pushed into the response collection for that calculated request ID.

This allows the plugin to support the case described in the issue. Where a later request for an entity results in a different response. 

### Related Issue(s)

Issue #128 

### Checklist

* [x] Follows [Contributing guidelines][1].
* [x] Pull Request title uses [Conventional Commit syntax][2].
* [x] All tests pass.
* [x] Linted code.
* [x] Authored new tests, if necessary.
* [x] Updated documentation, if necessary.

[1]:https://github.com/oreillymedia/cypress-playback/blob/main/CONTRIBUTING.md
[2]:https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by alaney** [comment]
@tvsbrent Pinging you on this to see if you have any feedback.

File: lib/commands/entities/PlaybackRequestMatcher.js [hunk 1]
```diff
@@ -195,9 +190,9 @@ class PlaybackRequestMatcher {
     this.#ignores = data.ignores;
     // Assume that a request being deserialized is stale.
     this.stale = true;
-    data.responses.forEach(entry => {
-      const response = new PlaybackResponse(entry);
-      this.#responses.set(response.id, response);
+    data.responseCollections.forEach(entry => {
+      const responseCollection = new PlaybackResponseCollection(entry);
+      this.#responseCollections.set(responseCollection.id, responseCollection);
```
Comment (tvsbrent) [inline lib/commands/entities/PlaybackRequestMatcher.js:195]: **suggestion:** We were discussing an approach that would keep us from having to introduce this as a breaking change. It's honestly something I should have thought of before, but here in the `deserialize` method for the `PlaybackRequestMatcher`, we could put in code to migrate the `responses` field into the `responseCollections` field.

I don't _think_ such a migration would be that difficult. (Although, I've been wrong more than once!) Some pseudo-code below:

```js
if (data.responses) {
  data.responses.forEach(entry => {
    const { id, url, ...rest } = entry;
    // This should match the shape that the response collection constructor/deserialize method expects
    const responseCollection = new PlaybackResponseCollection({ id, url, responses: [rest] });
    this.#responseCollection.set(responseCollection.id, responseCollection);
  });
} else {
  // Other deserialization code here.
  ...
}
```

https://github.com/oreillymedia/cypress-playback/blob/main/lib/commands/entities/PlaybackRequestMatcher.js#L198-L201

**Review by tvsbrent — COMMENTED** [review]
Thanks so much for this submission! We did have one suggestion that should make it easier to introduce this change.

**Comment by alaney** [comment]
Your suggestion seems to work. I implemented it and added a few tests.

**Review by tvsbrent — APPROVED** [review]

**Pull request merged**